Formula/kcptun.rb

```diff
@@ -0,0 +1,109 @@
+require "language/go"
+
+class Kcptun < Formula
+  desc "extremely simple & fast udp tunnel based on kcp protocol"
+  homepage "https://github.com/xtaci/kcptun"
+  url "https://github.com/xtaci/kcptun/archive/v20160701.tar.gz"
+  sha256 "865ad3dccd5bdd96075e7653da3b8ff4aaff576413c3b4e330d993e6f5209f68"
+  head "https://github.com/xtaci/kcptun.git"
+
+  depends_on "go" => :build
+
+  go_resource "golang.org/x/crypto" do
+    url "https://github.com/golang/crypto.git"
```

Comment:
These all need to be accompanied with a `revision`; see [`hugo.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/hugo.rb) for an example.